awswrangler/sqlserver.py

```diff
@@ -0,0 +1,368 @@
+    timeout: Optional[int] = 0,
+) -> "pyodbc.Connection":
+    """Return a pyodbc connection from a Glue Catalog Connection.
+
+    https://github.com/mkleehammer/pyodbc
+
+    Parameters
+    ----------
+    connection : Optional[str]
+        Glue Catalog Connection name.
+    secret_id: Optional[str]:
+        Specifies the secret containing the version that you want to retrieve.
+        You can specify either the Amazon Resource Name (ARN) or the friendly name of the secret.
+    catalog_id : str, optional
+        The ID of the Data Catalog.
+        If none is provided, the AWS account ID is used by default.
+    dbname: Optional[str]
+        Optional database name to overwrite the stored one.
+    odbc_driver_version : int
+        Major version of the OBDC Driver version that is installed and should be used.
+    boto3_session : boto3.Session(), optional
+        Boto3 Session. The default boto3 session will be used if boto3_session receive None.
+    timeout: Optional[int]
+        This is the time in seconds before the connection to the server will time out.
+        The default is None which means no timeout.
+        This parameter is forwarded to pyodbc.
+        https://github.com/mkleehammer/pyodbc/wiki/The-pyodbc-Module#connect
+
+    Returns
+    -------
+    pyodbc.Connection
+        pyodbc connection.
+
+    Examples
+    --------
+    >>> import awswrangler as wr
+    >>> con = wr.sqlserver.connect(connection="MY_GLUE_CONNECTION", odbc_driver_version=17)
+    >>> with con.cursor() as cursor:
+    >>>     cursor.execute("SELECT 1")
+    >>>     print(cursor.fetchall())
+    >>> con.close()
+
+    """
+    attrs: _db_utils.ConnectionAttributes = _db_utils.get_connection_attributes(
+        connection=connection, secret_id=secret_id, catalog_id=catalog_id, dbname=dbname, boto3_session=boto3_session
+    )
+    if attrs.kind != "sqlserver":
+        raise exceptions.InvalidDatabaseType(
+            f"Invalid connection type ({attrs.kind}. It must be a sqlserver connection.)"
+        )
+    connection_str = (
+        f"DRIVER={{ODBC Driver {odbc_driver_version} for SQL Server}};"
+        f"SERVER={attrs.host},{attrs.port};"
+        f"DATABASE={attrs.database};"
+        f"UID={attrs.user};"
+        f"PWD={attrs.password}"
+    )
+
+    return pyodbc.connect(connection_str, timeout=timeout)
+
+
+@_check_for_pyodbc
+def read_sql_query(
+    sql: str,
+    con: "pyodbc.Connection",
+    index_col: Optional[Union[str, List[str]]] = None,
+    params: Optional[Union[List[Any], Tuple[Any, ...], Dict[Any, Any]]] = None,
+    chunksize: Optional[int] = None,
+    dtype: Optional[Dict[str, pa.DataType]] = None,
+    safe: bool = True,
+) -> Union[pd.DataFrame, Iterator[pd.DataFrame]]:
+    """Return a DataFrame corresponding to the result set of the query string.
+
+    Parameters
+    ----------
+    sql : str
+        SQL query.
+    con : pyodbc.Connection
+        Use pyodbc.connect() to use "
```

Comment:
Typo?

Reply:
I think this should be fine. Or do you mean the package name or the formulation=

Reply:
Or do you think we should include `pymssql` here aswell, as it should be interchangeable?

Reply:
> Or do you think we should include pymssql here aswell, as it should be interchangeable?

No

I mean, the `con` description seems to be breaking line in a weird position and with some weird quotes.
Is this some pattern? 🤔

Reply:
It's actually just based on the postgresql module:
https://github.com/awslabs/aws-data-wrangler/blob/master/awswrangler/postgresql.py#L163

Reply:
Should be fixed in the latest commit.